src/main/java/seedu/address/model/expense/RecurringExpenseManager.java

```diff
@@ -134,10 +136,59 @@ public double getExpenseAmount() {
         return expenseAmount;
     }
 
+    public void setNextExpenseDate(LocalDate nextExpenseDate) {
+        this.nextExpenseDate = nextExpenseDate;
+    }
+
     @Override
     public String toString() {
         return "Recurring Expense: " + expenseName + " Amount: " + expenseAmount + " Category: "
                 + expenseCategory + " Start Date: " + startDate + " End Date: " + endDate
                 + " Recurring Expense Type: " + recurringExpenseType;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        RecurringExpenseManager recurringExpense = (RecurringExpenseManager) object;
+
+        if (!recurringExpense.startDate.isEqual(this.startDate)) {
+            return false;
+        }
+
+        if (recurringExpense.expenseAmount != this.expenseAmount) {
+            return false;
+        }
+
+        if (!recurringExpense.nextExpenseDate.isEqual(this.nextExpenseDate)) {
+            return false;
+        }
+
+        if (recurringExpense.recurringExpenseType != this.recurringExpenseType) {
+            return false;
+        }
+
+        if (!recurringExpense.expenseCategory.equals(this.expenseCategory)) {
+            return false;
+        }
+
+        if (!recurringExpense.expenseName.equals(this.expenseName)) {
+            return false;
+        }
+
+        if (recurringExpense.endDate == null) {
+            return this.endDate == null;
+        }
+
+        if (!recurringExpense.endDate.isEqual(this.endDate)) {
+            return false;
+        }
```

Comment:
this part might be a little repetitive, perhaps consider modifying this in the future to be a chain of && conditions instead and return true overall if all conditions are true.